### PR TITLE
[Wave Collect] [Xero] [Export Flow] Create the Preferred exporter select page

### PR DESCRIPTION
**Pull request by hungvu193**
### Details

Create the Preferred exporter select page
### Fixed Issues

$ https://github.com/Expensify/App/issues/39742
PROPOSAL: N/A

### Tests

Precondition: User connected to Xero
1. Go to Workspace => Accounting
2. Choose Export => Preferred Exporter, click on the MenuItem
3. Try to update the preferred exporter email few times.
4. Verify that it's updated correctly.
- [x] Verify that no errors appear in the JS console

### Offline tests

Same as Tests.
### QA Steps

Same as Tests.
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/16502320/11c5eb54-f44a-4402-afc5-6ea0f21f7378

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/16502320/12693b32-8eda-4cf6-b543-28c42465f0e4

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/16502320/47a87a0d-85bf-463b-89a2-323fd1bec20f

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/16502320/580d7ade-952c-46b9-8726-c746b59f6b5f

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/16502320/d12046d5-4c1e-4e91-b870-d7be9e28ad8f

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/16502320/25796372-227c-4194-a48a-325ee78f5aa3

</details>

### CODE REVIEW

**Comment by hungvu193** [comment]
Hold for https://github.com/Expensify/App/pull/41554

**Comment by mananjadhav** [comment]
This should be blocked for Invoice Account Selector PR as well right? I can see changes that I just reviewed.

File: src/libs/Navigation/AppNavigator/ModalStackNavigators/index.tsx [hunk 1]
```diff
@@ -1,42 +1,15 @@
-import type {ParamListBase} from '@react-navigation/routers';
-import type {StackNavigationOptions} from '@react-navigation/stack';
-import {createStackNavigator} from '@react-navigation/stack';
+import type { ParamListBase } from '@react-navigation/routers';
```
Comment (mananjadhav) [inline src/libs/Navigation/AppNavigator/ModalStackNavigators/index.tsx:1]: Do we need these changes related whitespace?

File: src/libs/Navigation/AppNavigator/ModalStackNavigators/index.tsx [hunk 2]
```diff
@@ -1,42 +1,15 @@
...
+import type { AddPersonalBankAccountNavigatorParamList, ChatFinderNavigatorParamList, DetailsNavigatorParamList, EditRequestNavigatorParamList, EnablePaymentsNavigatorParamList, FlagCommentNavigatorParamList, MoneyRequestNavigatorParamList, NewChatNavigatorParamList, NewTaskNavigatorParamList, ParticipantsNavigatorParamList, PrivateNotesNavigatorParamList, ProfileNavigatorParamList, ReferralDetailsNavigatorParamList, ReimbursementAccountNavigatorParamList, ReportDescriptionNavigatorParamList, ReportDetailsNavigatorParamList, ReportSettingsNavigatorParamList, RoomInviteNavigatorParamList, RoomMembersNavigatorParamList, SearchReportParamList, SettingsNavigatorParamList, SignInNavigatorParamList, SplitDetailsNavigatorParamList, TaskDetailsNavigatorParamList, TeachersUniteNavigatorParamList, WalletStatementNavigatorParamList, WorkspaceSwitcherNavigatorParamList } from '@navigation/types';
+import type { ThemeStyles } from '@styles/index';
+import type { Screen } from '@src/SCREENS';
 import SCREENS from '@src/SCREENS';
 import useModalScreenOptions from './useModalScreenOptions';
 import WorkspaceSettingsModalStackNavigator from './WorkspaceSettingsModalStackNavigator';
 
+
```
Comment (mananjadhav) [inline src/libs/Navigation/AppNavigator/ModalStackNavigators/index.tsx:12]: Is this needed?

File: src/libs/Navigation/AppNavigator/ModalStackNavigators/index.tsx [hunk 3]
```diff
@@ -384,4 +362,4 @@ export {
     ProcessMoneyRequestHoldStackNavigator,
     WorkspaceSettingsModalStackNavigator,
     SearchReportModalStackNavigator,
-};
+};
```
Comment (mananjadhav) [inline src/libs/Navigation/AppNavigator/ModalStackNavigators/index.tsx:365]: This one too?

File: src/pages/workspace/accounting/xero/export/XeroPreferredExporterSelectPage.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            if (row.value !== exportConfiguration?.exporter) {
+                Connections.updatePolicyConnectionConfig(policyID, CONST.POLICY.CONNECTIONS.NAME.XERO, CONST.XERO_CONFIG.EXPORT, {exporter: row.value});
+            }
+            Navigation.goBack(ROUTES.POLICY_ACCOUNTING_XERO_EXPORT.getRoute(policyID));
+        },
+        [policyID, exportConfiguration],
+    );
+
+    return (
+        <AccessOrNotFoundWrapper
```
Comment (mananjadhav) [inline src/pages/workspace/accounting/xero/export/XeroPreferredExporterSelectPage.tsx:58]: Use `ConnectionLayout`?

Comment (mananjadhav) [reply src/pages/workspace/accounting/xero/export/XeroPreferredExporterSelectPage.tsx:58]: I can see we also have a sub-note here? We could still use Connection Layout and only use the subnote in the SelectionList.

**Comment by hungvu193** [comment]
This PR is not ready yet. Please wait until I mark it as ready 😄

**Comment by mananjadhav** [comment]
Ohh 😄. I stumbled upon this one.

**Comment by hungvu193** [comment]
> Hold for #41554
Still being held but almost ready!

File: src/languages/en.ts [hunk 5]
```diff
@@ -2001,16 +2001,34 @@ export default {
...
                 autoSyncDescription: 'Sync Xero and Expensify automatically every day.',
-                purchaseBillStatusTitle: 'Set purchase bill status (optional)',
+                purchaseBillStatusTitle: 'Purchase bill status',
                 reimbursedReports: 'Sync reimbursed reports',
                 reimbursedReportsDescription: 'Any time a report is paid using Expensify ACH, the corresponding bill payment will be created in the Xero account below.',
                 xeroBillPaymentAccount: 'Xero Bill Payment Account',
                 xeroInvoiceCollectionAccount: 'Xero Invoice Collections Account',
+                invoiceAccountSelectorDescription: "As you've enabled exporting invoices from Expensify to Xero, this is the account the invoice will appear against once marked as paid.",
             },
+            exportPreferredExporterNote: 'This can be any workspace admin, but must be a Domain Admin if you set different export accounts for individual company cards in Domain Settings.',
```
Comment (mananjadhav) [inline src/languages/en.ts:2030]: We've used small case for `workspace admin` and title case for `Domain Admin`. Let's confirm this once,

File: src/pages/workspace/accounting/xero/export/XeroPreferredExporterSelectPage.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+            return [
+                {
+                    value: policyOwner,
+                    text: policyOwner,
+                    keyForList: policyOwner,
+                    isSelected: true,
+                },
+            ];
+        }
+        return exporters?.reduce<CardListItem[]>((vendors, vendor) => {
```
Comment (mananjadhav) [inline src/pages/workspace/accounting/xero/export/XeroPreferredExporterSelectPage.tsx:35]: Can we replace this reduce with:
```
 return exporters
        ?.filter(vendor => vendor.email) // Filter out vendors without emails
        .map(vendor => ({
            value: vendor.email,
            text: vendor.email,
            keyForList: vendor.email,
            isSelected: exportConfiguration?.exporter === vendor.email,
        })) ?? []

```

**Review by mananjadhav — COMMENTED** [review]
Small refactor comments, but I think we should test this once Export page is merged.

File: src/pages/workspace/accounting/xero/export/XeroPreferredExporterSelectPage.tsx (hunk 6 above)
Comment (hungvu193) [reply src/pages/workspace/accounting/xero/export/XeroPreferredExporterSelectPage.tsx:35]: I'd do a `reduce` rather than combining `map` and `filter` 😄

**Review by mananjadhav — APPROVED** [review]

**Comment by mananjadhav** [comment]
Tagging @Expensify/design. I wanted to also confirm if it's the only email in the list do we need to navigate twice?

**Comment by dannymcclain** [comment]
Thanks for bringing this up—no, we do not want to do the double-navigate thing you're showing. The second screen should more closely follow what we do for language select (for example). That would give us something like this: 
CleanShot 2024-05-07 at 16 39 20@2x

Where we have the push input on the first screen, and then the immediate screen after that is the list with the explainer text.

File: src/pages/workspace/accounting/xero/export/XeroExportConfigurationPage.tsx [hunk 7]
```diff
@@ -22,7 +24,9 @@ function XeroExportConfigurationPage({policy}: WithPolicyConnectionsProps) {
     const menuItems: MenuItem[] = [
         {
             description: translate('workspace.xero.preferredExporter'),
-            onPress: () => {},
+            onPress: () => {
```
Comment (mananjadhav) [inline src/pages/workspace/accounting/xero/export/XeroExportConfigurationPage.tsx:27]: @hungvu193 @lakchote I am not sure how we're handling the pendingFields here. I can see it's tagged as `policy.pendingFields.xeroConfigPreferredExporter` in the document but I don't see it being used. Neither here nor in @lakchote's PR for Bill selector.

**Comment by trjExpensify** [comment]
> Thanks for bringing this up—no, we do not want to do the double-navigate thing you're showing. The second screen should more closely follow what we do for language select (for example). That would give us something like this: CleanShot 2024-05-07 at 16 39 20@2x
> 
> Where we have the push input on the first screen, and then the immediate screen after that is the list with the explainer text.

Huge +1.

**Comment by dubielzyk-expensify** [comment]
Exactly. Thanks for clarifying, @dannymcclain . I assume this is reflected in the Figma file? If not, let's update it

**Comment by hungvu193** [comment]
I did it like the docs: https://docs.google.com/document/d/1VEIzxwNzedLNestAFVT5InGvD4N6FHb0922xjMGOkko/edit#heading=h.fbu7ml85rrhl
Can we update the docs?

**Comment by shawnborton** [comment]
Big agree with where you landed Danny.

**Comment by hungvu193** [comment]
Cool. Let's me update it then.

**Comment by hungvu193** [comment]
Updated!

https://github.com/Expensify/App/assets/16502320/9f544b6b-3444-4cc8-a0ba-7e9d4239fabb

**Comment by lakchote** [comment]
cc @Expensify/design does that look good to you too now?

**Comment by dubielzyk-expensify** [comment]
That looks right to me!

**Comment by mananjadhav** [comment]
@lakchote @hungvu193 I've raised another question related to pendingFields.

File: src/pages/workspace/accounting/xero/export/XeroExportConfigurationPage.tsx (hunk 7 above)
Comment (hungvu193) [reply src/pages/workspace/accounting/xero/export/XeroExportConfigurationPage.tsx:27]: Yeah, I didn't find any usage of it.

Comment (lakchote) [reply src/pages/workspace/accounting/xero/export/XeroExportConfigurationPage.tsx:27]: Yes, it shouldn't be `xeroConfigPreferredExporter` or anything like that. I previously wrote that before the implementation and I was wrong.

The correct way to do this is would be here to use `pendingFields?.export` for the pending action.

**Review by mananjadhav — APPROVED** [review]
Reviewed again and uploaded revised screenshots in the checklist.

**Comment by hungvu193** [comment]
@lakchote This will be ready for your final review 👀

**Review by lakchote — APPROVED** [review]

**Comment by lakchote** [comment]
> @lakchote This will be ready for your final review 👀

Approved, conflicts to be solved and then I'll merge it @hungvu193 👍

**Comment by hungvu193** [comment]
Cool! I've resolved the conflicts

**Comment by hungvu193** [comment]
I've fixed the lint as well, this will be ready to merge 😄

**Pull request merged**